addons/sale_crm/tests/test_crm_lead_to_quotation_salesperson.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.crm.tests import common as crm_common
+from odoo.tests.common import Form
+
+
+class TestLeadToQuotationSalesperson(crm_common.TestCrmCases):
+
+    def test_sale_quotation_from_crm_lead(self):
+        # Test that correct sales person and team on quotation is set when created from lead
+
+        CrmLead = self.env['crm.lead']
+        partner_1 = self.env['res.partner'].create({
+            'name': 'Partner 1',
+            'email': 'partner@other.company.com'
+        })
+        partner_2 = self.env['res.partner'].create({
+            'name': 'Partner 2',
+            'email': 'supplier@other.company.com',
+            'user_id': self.crm_salemanager.id,
+        })
+
+        # Check that if lead has a salesperson and lead contact also has a salesperson, lead's
+        # salesperson should be set as a quotation's salesperson (instead of one linked 
+        # with lead contact) while creating quotation.
+        lead_without_salesperson_1 = CrmLead.create({
+            'name': 'Lead 2',
+            'partner_id': partner_2.id,
+            'user_id': False
+            })
+        action_ctx = lead_without_salesperson_1.action_new_quotation().get('context', {})
+        quotation_1 = Form(self.env['sale.order'].with_context(action_ctx)).save()
+        self.assertEqual(partner_2.user_id.id, quotation_1.user_id.id, 'quotation should have the same salesperson as lead')
+        # checks quotation should have the same sales team as lead
+        self.assertEqual(lead_without_salesperson_1.team_id.id, quotation_1.team_id.id, 'quotation should have the same sales team as lead')
```

Comment:
Note that you can compare directly recordsets (and therefore remove .id / .ids), aka
```
self.assertEqual(lead_without_salesperson_1.team_id, quotation_1.team_id, '...')
```